app/controllers/spree/admin/reports_controller_decorator.rb

```diff
@@ -181,20 +189,37 @@ def sales_tax
   end
 
   def bulk_coop
-    # -- Prepare date parameters
-    prepare_date_params params
+    if request.format.json?
```

Comment:
Ok, cool. I am happy to take that on as our preferred approach from now on if others are. Next question: what namespacing and routing do we use for these for new API controllers? I don't really like the idea of just mashing them into Spree's existing API controller structure, but maybe I just need to get over it...